File: 02_스택, 큐, 덱/라이브 코딩/10845.cpp
// ť
#include <iostream>
#include <queue>

using namespace std;

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);
	
	int n, x;
	queue<int> s;
	string input;

	cin >> n;
	while (n--) {
		cin >> input;
		if (input == "push") {
			cin >> x;
			s.push(x);
		}
		else if (input == "pop") {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.front() << '\n';
				s.pop();
			}
		}
		else if (input == "size") {
			cout << s.size() << '\n';
		}
		else if (input == "empty") {
			cout << s.empty() << '\n';
		}
		else if (input == "front") {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.front() << '\n';
			}
		}
		else if (input == "back") {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.back() << '\n';
			}
		}
	}
}

File: 02_스택, 큐, 덱/라이브 코딩/10845.py
from collections import deque
N = int(input())
que = deque()
ans = ""

for _ in range(N):
    inp = input().split()
    command = inp[0]

    if command == 'push':
        que.append(int(inp[1]))
    elif command == 'pop':
        if len(que) == 0:
            ans += str(-1) + '\n'
        else:
            ans += str(que.popleft()) + '\n'
    elif command == 'size':
        ans += str(len(que)) + '\n'
    elif command == 'empty':
        if len(que) == 0:
            ans += str(1) + '\n'
        else:
            ans += str(0) + '\n'
    elif command == 'front':
        if len(que) == 0:
            ans += str(-1) + '\n'
        else:
            ans += str(que[0]) + '\n'

    else:
        if len(que) == 0:
            ans += str(-1) + '\n'
        else:
            ans += str(que[-1]) + '\n'

print(ans)


